core/controllers/reader.py

```diff
@@ -281,12 +284,33 @@ def get(self, exploration_id):
                     'data_schema_version': data_schema_version
                 }
 
+        pretest_question_dicts = []
+        if story_id:
+            story = story_services.get_story_by_id(story_id, strict=False)
+            if story is None:
+                raise self.InvalidInputException
+
+            if (
+                    story.get_prerequisite_skill_ids_for_exp_id(exploration_id)
```

Comment:
I thnk this should be story.has_exploration(exploration_id) or similar.

Reply:
Done.